pulp_rpm/app/tasks/publishing.py

```diff
@@ -566,36 +566,49 @@ def generate_repo_metadata(
             oth_db.add_pkg(pkg)
 
     # Process update records
-    for update_record in UpdateRecord.objects.filter(pk__in=content).iterator():
+    update_records = UpdateRecord.objects.filter(pk__in=content).order_by("id", "digest")
+    for update_record in update_records.iterator():
+        if not upd_xml:
```

Comment:
can you move this check before forloop?

Reply:
I could but we would have to do the query twice, once to see if any updaterecords exist. This way only needs to perform it once.  I don't think it actually would make a difference in practice so if you prefer the other way I can do that

Reply:
no this is good as is :+1:

Reply:
is it worth adding a changelog line about not publishing empty updateinfo anymore?